connect/eaas/runner/managers/transformation.py

```diff
@@ -0,0 +1,282 @@
+        :param tfn_request: Connect tfn request that need to be processed.
+        :type tfn_request: dict
+        """
+
+        future = asyncio.create_task(self.process_transformation(
+            task_data, tfn_request, method,
+        ))
+
+        logger.info(f'Enqueue result for task {task_data.options.task_id}')
+        asyncio.create_task(self.enqueue_result(task_data, future))
+
+    async def build_response(self, task_data, future):
+        result_message = Task(**task_data.dict())
+        try:
+            begin_ts = time.monotonic()
+            result = await asyncio.wait_for(
+                future,
+                timeout=self.config.get_timeout('transformation'),
+            )
+            result_message.output = TaskOutput(result=result.status)
+            result_message.output.runtime = time.monotonic() - begin_ts
+            logger.info(
+                f'Transformation task {task_data.options.task_id} '
+                f'result: {result.status}, '
+                f'took: {result_message.output.runtime}',
+            )
+            if result.status in (ResultType.SKIP, ResultType.FAIL):
+                result_message.output.message = result.output
+        except Exception as e:
+            self.log_exception(task_data, e)
+            result_message.output = TaskOutput(result=ResultType.RETRY)
+            result_message.output.message = traceback.format_exc()[:4000]
+
+        return result_message
+
+    async def process_transformation(self, task_data, tfn_request, method):
+        download_result = await asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.download_exel,
+            tfn_request,
+        )
+        input_file = download_result.reslut()
+        output_file = NamedTemporaryFile(
+            suffix=tfn_request['files']['input']['name'].split('.')[-1],
+        )
+
+        read_queue = asyncio.Queue(TRANSFORMATION_TASK_MAX_PARALLEL_LINES)
+        write_queue = asyncio.Queue()
+
+        reader_task = asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.read_excel,
+            input_file.name,
+            read_queue,
+        )
+        writer_task = asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.write_excel,
+            output_file.name,
+            write_queue,
+            tfn_request['stats']['total'],
+            task_data.input.object_id,
+        )
+        processor_task = asyncio.create_task(self.process_rows(
+            read_queue, write_queue, method, total_rows,
+        ))
+
+        try:
+            await asyncio.gather([reader_task, writer_task, processor_task])
+            self.send_output_file(task_data, input_file)
+            input_file.close()
+        except Exception as e:
+            self.log_exception(task_data, e)
+            raise
+
+    def download_excel(self, tfn_request):
+        client = httpx.AsyncClient()
```

Comment:
This function is sync because the file operation is sync. So you should use requests here.